book/src/dev/rfcs/drafts/0005-treestate.md

```diff
@@ -177,11 +177,29 @@ state. The Sprout and Sapling nullifiers revealed in the block will be merged
 with the exising ones in our finalized state (ie, it should strictly grow over
 time).
 
-IMPORTANT: we need to save the incremental merkle tree / serialized nodes for:
+## State Management
+
+### Orchard
+- There is a single copy of the latest Orchard Note Commitment Tree for the finalized tip.
+- When finalizing a block, the finalized tip is updated with a serialization of the latest Orchard Note Commitment Tree. (The previous tree should be deleted as part of the same database transaction.)
+- Each non-finalized chain gets its own copy of the Orchard note commitment tree, cloned from the note commitment tree of the finalized tip or fork root.
+- When a block is added to a non-finalized chain tip, the Orchard note commitment tree is updated with the note commitments from that block.
+- When a block is rolled back from a non-finalized chain tip... (TODO)
```

Comment:
> When a block is rolled back from a non-finalized chain tip... (TODO)

I'm just going to tag this TODO with a comment, so it doesn't get lost.

I assume we'll do the same thing for sapling.